### PR TITLE
API: add CSSStyleDeclaration

### PR DESCRIPTION
**Pull request by caugner**
Sources:
1. CSS Object Model specification:
   https://drafts.csswg.org/cssom/#the-cssstyledeclaration-interface
2. Chromium repository:
   https://github.com/chromium/chromium/blob/1182c84d9849940a6f6be3ec22d8b244dc554a53/third_party/blink/renderer/core/css/css_style_declaration.idl
   https://chromium.googlesource.com/chromium/src/+/ca8687a29947918259b06067dbc6b412582c4c3c%5E%21/third_party/WebKit/WebCore/css/CSSStyleDeclaration.idl
3. Firefox repository:
   https://github.com/mozilla/gecko-dev/blob/04b9cbbc2be2137a37e158a5ebaf9c7bef2364f9/dom/webidl/CSSStyleDeclaration.webidl
   https://github.com/mozilla/gecko-dev/commit/7cff1485c880669fcaa2041b81574c45e6392753#diff-3ba774067d5fc9b261902177c4a0aa0d

### CODE REVIEW

File: api/CSSStyleDeclaration.json [hunk 1]
```diff
@@ -0,0 +1,484 @@
...
+          }
+        },
+        "status": {
+          "experimental": false,
+          "standard_track": true,
+          "deprecated": false
+        }
+      }
+    },
+    "cssText": {
```
Comment (connorshea) [inline api/CSSStyleDeclaration.json]: This isn't a sub-element of CSSStyleDeclaration (neither are any of the other subfeatures), the indentation needs to be fixed  

The file with code folding:

screen shot 2018-06-09 at 3 40 37 pm

**Comment by connorshea** [comment]
Also worth noting that the MDN articles for each of the subfeatures have browser compat tables (the main feature article does not, for whatever reason), e.g. https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleDeclaration/setProperty#Browser_compatibility

I assume since you researched these, the old tables probably don't matter, but I figured I'd mention it :)

File: api/CSSStyleDeclaration.json (hunk 1 above)
Comment (connorshea) [reply api/CSSStyleDeclaration.json]: After a quick check I _think_ this is the only PR you did this on.

Comment (caugner) [reply api/CSSStyleDeclaration.json]: Thanks and sorry, fixed now in 33b2a3d.

**Comment by caugner** [comment]
> I assume since you researched these, the old tables probably don't matter, but I figured I'd mention it :)

@connorshea Thank you for mentioning that, I was in fact not really considering that the individual feature articles may have more compat data than the main article. 🤦

**Comment by connorshea** [comment]
@caugner I think it's a weird edge case, I haven't seen it anywhere else ¯\_(ツ)_/¯

**Comment by connorshea** [comment]
I started on this without realizing you had already done it :D 

This PR is missing the [getPropertyCSSValue](https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleDeclaration/getPropertyCSSValue) method. I know it's obsolete, but I'm not sure if it should be excluded from the dataset?

**Comment by caugner** [comment]
> This PR is missing the getPropertyCSSValue method. I know it's obsolete, but I'm not sure if it should be excluded from the dataset?

According to [MDN](https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleDeclaration/getPropertyCSSValue#Specifications), it is deprecated since 2003, but I don't mind adding it.

**Comment by caugner** [comment]
> Also worth noting that the MDN articles for each of the subfeatures have browser compat tables [...]

I have added that data now (d427fe6e836144a9b180bf375d263bfec467c98b), mainly for IE/Edge/Safari.

**Comment by ddbeck** [comment]
Quite happy with this, thanks @caugner for the PR and @connorshea for the review! 🎉

**Pull request merged**